apps/tailwind-components/components/input/Listbox.vue

```diff
@@ -0,0 +1,345 @@
+<template>
+  <div class="w-full relative">
+    <button
+      :id="id"
+      ref="listbox-button"
+      role="combobox"
+      :disabled="disabled"
+      aria-haspopup="listbox"
+      :aria-controls="`listbox-${id}-options`"
+      :aria-required="required"
+      :aria-expanded="isExpanded"
+      :aria-labelledby="labelId"
+      class="flex justify-start items-center h-10 w-full text-left pl-10 border bg-input rounded-search-input text-button-input-toggle focus:ring-blue-300"
+      :class="{
+        'border-disabled text-disabled bg-disabled': disabled,
+        'border-invalid text-invalid': hasError,
+      }"
+      @click.prevent="openCloseListbox"
+      @keydown.prevent="onListboxButtonKeyDown"
+    >
+      <span class="w-full">
+        {{ listboxDisplayText }}
+      </span>
+      <div class="w-[60px] flex flex-col">
+        <BaseIcon :width="18" name="caret-up" class="mx-auto -my-1" />
+        <BaseIcon :width="18" name="caret-down" class="mx-auto -my-1" />
+      </div>
+    </button>
+
+    <ul
+      :id="`listbox-${id}-options`"
+      role="listbox"
+      ref="listbox-ul"
+      :aria-expanded="isExpanded"
+      class="absolute b-0 w-full overflow-y-scroll z-10 bg-listbox"
+      :class="{
+        hidden: !isExpanded,
+        'h-44': isExpanded && listboxOptions.length > 5,
+        'shadow-inner': isExpanded,
+      }"
+      @keydown.prevent="onListboxKeyDown"
+    >
+      <li
+        v-for="option in listboxOptions"
+        ref="listbox-li"
+        :id="option.elemId"
+        role="option"
+        class="flex justify-start items-center gap-3 pl-3 py-1 text-listbox border-t-[1px] border-t-listbox-option hover:cursor-pointer hover:bg-listbox-hover hover:text-listbox focus:bg-listbox-hover focus:text-listbox"
+        :class="{
+          '!bg-listbox-selected !text-listbox-selected':
+            option.value === (modelValue as IListboxOption)?.value || option.value === modelValue,
+        }"
+        :aria-selected="option.value === (modelValue as IListboxOption)?.value || option.value === modelValue"
```

Comment:
move logic to a function?

Reply:
If we add support for objects as value, the === needs to be replaced with a more complex comperator.

Reply:
Done.